zsh-copilot.plugin.zsh

```diff
@@ -69,57 +64,67 @@ function _suggest_ai() {
     #     fi
     # fi
 
-    local data="{
-            \"model\": \"gpt-4\",
-            \"messages\": [
-                {
-                    \"role\": \"system\",
-                    \"content\": \"$PROMPT\"
-                },
-                {
-                    \"role\": \"user\",
-                    \"content\": \"$input\"
-                }
-            ]
-        }"
-    local response=$(curl "https://${OPENAI_API_URL}/v1/chat/completions" \
-        --silent \
-        -H "Content-Type: application/json" \
-        -H "Authorization: Bearer $OPENAI_API_KEY" \
-        -d $data)
-    local message=$(echo "$response" | jq -r '.choices[0].message.content')
-
-    # zle -U "$suggestion"
+    # If Ollama is configured, use it; otherwise, fallback to OpenAI
+    if [[ -n "$ZSH_COPILOT_KEY_OLLAMA" ]]; then
+        zle -R "Thinking (Ollama)..."
+        
+        # Call Ollama using llama3 model
+        local response=$(echo "$PROMPT $input" | ollama run llama3.1)
```

Comment:
I think someone recommended using a http based approach instead, please check that again